how-to/how-to-use-virtio-mem-with-kata.md

```diff
@@ -0,0 +1,49 @@
+# Kata Containers with virtio-mem
+
+- [Introduction](#introduction)
+- [Pre-requisites](#pre-requisites)
+- [Run a Kata Container with utilizing virtio-mem](#run-a-kata-container-utilizing-virtio-mem)
+
+## Introduction
+
+The basic idea of virtio-mem is to provide a flexible, cross-architecture memory hot(un)plug solution that avoids many limitations imposed by existing technologies, architectures, and interfaces. More details can be found in https://lkml.org/lkml/2019/12/12/681.
+
+Kata Containers with virtio-mem support memory resize.
+
+## Pre-requisites
+
+Kata Containers with virtio-mem requires Linux and the qemu that support virtio-mem.
+
+The Linux kernel is at https://github.com/davidhildenbrand/linux/tree/virtio-mem-rfc-v4.
+The Linux kernel config that can work with Kata Containers is at https://gist.github.com/teawater/016194ee84748c768745a163d08b0fb9.
+
+The qemu is at https://github.com/teawater/qemu/tree/kata-virtio-mem. (The original source is at https://github.com/davidhildenbrand/qemu/tree/virtio-mem.  Its base version of qemu cannot work with Kata Containers.  So merge the commit of virtio-mem to upstream qemu.)
```

Comment:
Instead of having this in a personal repo, maybe we can have this as a separate branch in the qemu repo under kata-containers project. cc @jcvenegas 
Same for the kernel.

Reply:
Do you know the expected time to be merged in upstream?

Reply:
> Do you know the expected time to be merged in upstream?

@davidhildenbrand Could you help me with this question?

Reply:
I'm planning to post v1 (as follow up to RFC v4) of the Linux guest driver side soon. Depending on the review feedback (and the missing ACK for a mm patch), I guess this could get merged fairly soon.

I'm currently working on the QEMU side, which is an early RFC. A very basic version could be posted/upstream soon, however, I am looking into implementing some reasonable improvements (e.g., resizeable memory regions) before sending out a v1. My goal is to send a v1 within the next months.

As always with upstream work, it's hard to make estimates :)

Reply:
I opened a new PR https://github.com/kata-containers/runtime/issues/2485 to discussion about add special branch for virtio-mem in Linux and qemu.